### PR TITLE
Adds focus trap to lightbox.js

### LINKED ISSUE #4460: Issue ID: DAC_Magnified_Images1 - Feature: Gallery
Report link: https://drive.google.com/file/d/1myaQdkLNCNhmrIoe_BPRwOxvpPxvT7w0/view
Full issue list: https://docs.google.com/spreadsheets/d/1vWQtZfD49uq7uNCNELuYN_HcIj-BKiE4t4Hwda3tJaQ/edit#gid=0

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

- changes container for lightbox from `div` => `dialog` to get all the nice focus trap functionality from that. A nice thing we learnt was you can use `showModal` or `show`. `showModal` set the focus trap, `show` doesn't.
- We've set the `tabindex` on the share buttons to `-1` as _all_ the images are included in the DOM of the `dialog`, along with their share buttons, see the video for details. This might not be a long term solution, but it works until we revisit this.

## being able to tab into slides without loaded images
https://user-images.githubusercontent.com/31692/173605588-7ae98d64-5a18-4860-a33d-d1fb1604b60c.mov

## being focus trapped 
https://user-images.githubusercontent.com/31692/173605658-bf0a4387-8a92-49db-9baa-7440a09f99ae.mov

## Alternatives we tried
- We tried setting non-selected slides to hidden, this doesn't work as we need to support swiping. The currently implementation requires that the DOM is not hidden.
- Use the [`inert`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert) attribute on the non-selected slides, which worked, but decided against it due to [lack of device support](https://caniuse.com/mdn-api_htmlelement_inert).

Closes https://github.com/guardian/dotcom-rendering/issues/4460

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/gallery/lightbox.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ class GalleryLightbox {
             });
         };
 
-        const galleryLightboxHtml = `<div class="overlay gallery-lightbox gallery-lightbox--closed gallery-lightbox--hover">
```
Comment (oliverlloyd) [inline static/src/javascripts/projects/common/modules/gallery/lightbox.js:59]: What does the removal of `gallery-lightbox--closed` do, was this just unused?

File: static/src/javascripts/projects/common/modules/gallery/lightbox.js [hunk 2]
```diff
@@ -357,6 +357,11 @@ class GalleryLightbox {
     }
 
     show() {
+        // `showModal` ensures that we focus trap the user within the `dialogue` container
+        // visually show/hide functionality is still controlled via css and classnames
+        // in the `this.hide` method, we close the modal
```
Comment (oliverlloyd) [inline static/src/javascripts/projects/common/modules/gallery/lightbox.js:362]:
```suggestion
        /**
         * Using `showModal` ensures that we trap focus inside the modal `dialogue`
         * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/showModal}
         *
         * Visually, show/hide functionality is still controlled via css and classnames in the `this.hide` method
         */
```

**Review by oliverlloyd — APPROVED** [review]
Nice solution!

We should also adopt this in the [abuse report form over in Discussion](https://github.com/guardian/discussion-rendering/blob/9655e9c9ef4976898471b95f93aefa89fa777789/src/components/AbuseReportForm/AbuseReportForm.tsx)

**Comment by OllysCoding** [comment]
Closes https://github.com/guardian/dotcom-rendering/issues/4460

File: static/src/javascripts/projects/common/views/content/block-sharing.html [hunk 3]
```diff
@@ -12,3 +12,4 @@
             <%=shareButtons%>
         </div>
 </li>
+=
```
Comment (bryophyta) [inline static/src/javascripts/projects/common/views/content/block-sharing.html:15]: Just wanted to check whether this line was supposed to have been added? (I'm not familiar with this templating markup style.)

File: static/src/javascripts/projects/common/modules/gallery/lightbox.js (hunk 2 above)
Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/gallery/lightbox.js:362]: Yes, JSDoc everywhere!

File: static/src/javascripts/projects/common/modules/gallery/lightbox.js (hunk 1 above)
Comment (jamesgorrie) [reply static/src/javascripts/projects/common/modules/gallery/lightbox.js:59]: I think it was actually by error. Not too sure what it does, but we'll leave it there as there's no tests, and this state machine is pretty complex.

**Pull request merged**